src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -102,21 +102,21 @@ private fun Class<out Any>.getMethod(name: String, parameterTypes: Array<Class<o
  */
 internal fun createLincheckResult(res: Any?, wasSuspended: Boolean = false) = when {
     (res != null && res.javaClass.isAssignableFrom(Void.TYPE)) || res is Unit -> if (wasSuspended) SuspendedVoidResult else VoidResult
-    res != null && res is Throwable -> ExceptionResult.create(res, wasSuspended)
+    res != null && res is Throwable -> ExceptionResult.create(res)
     res === COROUTINE_SUSPENDED -> Suspended
     res is kotlin.Result<Any?> -> res.toLinCheckResult(wasSuspended)
-    else -> ValueResult(res, wasSuspended)
+    else -> ValueResult(res)
 }
 
 private fun kotlin.Result<Any?>.toLinCheckResult(wasSuspended: Boolean) =
     if (isSuccess) {
         when (val value = getOrNull()) {
             is Unit -> if (wasSuspended) SuspendedVoidResult else VoidResult
             // Throwable was returned as a successful result
-            is Throwable -> ValueResult(value::class.java, wasSuspended)
-            else -> ValueResult(value, wasSuspended)
+            is Throwable -> ValueResult(value::class.java)
```

Comment:
Shouldn't it be ExceptionResult?

Reply:
Nope, this is a case when Throwable was returned as a successful result, see comment above

Reply:
Why should you extract a class in one case and a value in another? `ExceptionResult` should be a successful one as well.

Reply:
I suppose the current code is (and was) buggy here.

Reply:
Ok, I agree, let's treat this case as a regular one, creating `ValueResult(value)` here.